### PR TITLE
Tutorial navigation improvement

### LINKED ISSUE #2547: 01_basic Tutorial Navigation Improvement
Many of the lessons in the [K Tutorial](https://kframework.org/k-distribution/k-tutorial/1_basic/) refer to previous lessons. For example, in [lesson 18](https://kframework.org/k-distribution/k-tutorial/1_basic/18_equality_and_conditionals/), exercise 2, there is a back-reference to lesson 16:

> Modify your solution to Lesson 1.16...

However, on the tutorial site hosted at [kframework.org](https://kframework.org/k-distribution/k-tutorial/1_basic/), as opposed to the source code on GitHub, there is no easy way to navigate back to earlier lessons.

Note: I'm just scoping this issue to the [01_basic](https://github.com/runtimeverification/k/tree/master/k-distribution/k-tutorial/1_basic) tutorial sections. Depending on how this turns out, my thinking is that other issues can be opened for the [intermediate](https://github.com/runtimeverification/k/tree/master/k-distribution/k-tutorial/2_intermediate) and [advanced](https://github.com/runtimeverification/k/tree/master/k-distribution/k-tutorial/3_advanced) sections if necessary.

Potential Solutions
-----------------------------------

These are not mutually-exclusive. Any subset of them could be done together:

1. Add a navigation tree or list in the sidebar of the [tutorial page](https://kframework.org/k-distribution/k-tutorial/1_basic/) allowing a user to expand and navigate to whatever page they like. For example, perhaps this [sidebar.html](https://github.com/runtimeverification/k/blob/382a862f0fe5fd1ca00c4c5d187939db2b4b0506/web/static_content/html/include/sidebar.html) file could be modified to include collapsible content under the various headers.
1. Add "Previous lesson" links to all the lessons in the tutorial - I'd be happy to do this and open a PR.
1. Whenever a previous lesson is referenced from another lesson in the tutorial, link to that lesson explicitly - I'd be happy to do this and open a PR (although I couldn't promise to catch 100% of the back-links).

Potential Alternatives/Work-arounds
-----------------------------------

1. If you work through the tutorial directly from GitHub or from the same repository cloned locally, instead of from kframework.org, then you can navigate through the tutorial lessons easily by navigating from the parent directory.
1. If you navigate through the lessons on kframework.org, and need to go back to an earlier lesson, as a workaround you can always visit the top-level [GitHub page](https://github.com/runtimeverification/k/tree/master/k-distribution/k-tutorial/1_basic) for the tutorial, then find out what a previous lesson is called, then type it into the URL in the address bar following "https://kframework.org/k-distribution/k-tutorial/1_basic/". This is tedious though.

### PR DESCRIPTION
**Pull request by shd101wyy**
Sandbox website: https://shd101wyy.github.io/k/

Fixes: https://github.com/runtimeverification/k/issues/2547

@ehildenb @radumereuta I have updated the left sidebar toc. It now also has a tree view. However, we will have to manually set the ToC. Right now I defined it in the [toc.md](https://github.com/shd101wyy/k/blob/doc/tutorial-navigation-improvement/web/toc.md) file. 

I also build the website into `.pdf`, `.html`, `.epub` and `.mobi` files by the [toc.md](https://github.com/shd101wyy/k/blob/doc/tutorial-navigation-improvement/web/toc.md) file. You could download them on the top right corner dropdown.

Screenshot from 2022-05-27 22-20-53

I will continue next monday. Still needs to fix a few bugs. Please let me know if I need to make any changes.

Thanks!

### CODE REVIEW

**Comment by ehildenb** [comment]
It looks good to me, I want sign-off from @radumereuta and @dwightguth though as well, so that they know now that _the table-of-contents for the tutorial is manually written here!_. You must update it if you update the tutorial.

**Review by radumereuta — COMMENTED** [review]
I haven't written too much of the documentation, but I will keep in mind the new structure.
I will let @dwightguth give the final approval.

File: web/toc.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    - [Lesson 1.12: Syntactic Lists](/k-distribution/k-tutorial/1_basic/12_syntactic_lists/README.md)
+    - [Lesson 1.13: Basics of K Rewriting](/k-distribution/k-tutorial/1_basic/13_rewrite_rules/README.md)
+    - [Lesson 1.14: Defining Evaluation Order](/k-distribution/k-tutorial/1_basic/14_evaluation_order/README.md)
+    - [Lesson 1.15: Configuration Declarations and Cell Nesting](/k-distribution/k-tutorial/1_basic/15_configurations/README.md)
+    - [Lesson 1.16: Maps, Semantic Lists, and Sets](/k-distribution/k-tutorial/1_basic/16_collections/README.md)
+    - [Lesson 1.17: Cell Multiplicity and Cell Collections](/k-distribution/k-tutorial/1_basic/17_cell_multiplicity/README.md)
+    - [Lesson 1.18: Term Equality and the Ternary Operator](/k-distribution/k-tutorial/1_basic/18_equality_and_conditionals/README.md)
+    - [Lesson 1.19: Debugging with GDB](/k-distribution/k-tutorial/1_basic/19_debugging/README.md)
+    - [Lesson 1.20: K Backends and the Haskell Backend](/k-distribution/k-tutorial/1_basic/20_backends/README.md)
+    - [Lesson 1.21: Unification and Symbolic Execution](/k-distribution/k-tutorial/1_basic/21_symbolic_execution/README.md)
```
Comment (dwightguth) [inline web/toc.md:42]: We should probably only include the lessons that are actually done here, to avoid confusion.

File: Dockerfile [hunk 2]
```diff
@@ -27,6 +27,11 @@ RUN    apt-get update        \
         python3-graphviz     \
         zlib1g-dev
 
+RUN     apt-get update       \
+    && apt-get install --yes \
+        texlive-xetex        \
+        pandoc
+
```
Comment (ehildenb) [inline Dockerfile]: @shd101wyy please add this to the prior list above, not make a new entry.

Comment (shd101wyy) [reply Dockerfile]: @ehildenb Done. I am not sure what to remove in `toc.md` though. Should we keep it just like that in this PR? Thanks

File: web/toc.md (hunk 1 above)
Comment (shd101wyy) [reply web/toc.md:42]: Hi @dwightguth . I have removed the unimplemented lessons. Please help review the PR again when you have time. Thanks!

**Review by dwightguth — CHANGES_REQUESTED** [review]
The code looks okay but you screwed up the git history, including a commit you shouldn't. Strip it out and I'll approve.

**Comment by shd101wyy** [comment]
Hi @dwightguth . I just rebased my commits. Thanks!

**Review by dwightguth — APPROVED** [review]

**Pull request merged**